### ISSUE TITLE
Simplify grammar for f-strings and s-strings

### ISSUE THREAD

**Issue by aljazerzen** [PRQL/prql#1828]
## What we have

Currently we allow all these things:
```
f"Hello my name is {name}"
f"Hello my name is {obj.get('name')}"
f"Hello my name is {obj.get("name")}"
f"Hello my name is {"this is all {just} a string"}"
f"Hello my name is { f"a {nested} f-string" }"
f"Hello my name is { f"a { f"double {nested}" f-string" }"
f"Evaluates to single braces {{name}}"
```

There is a case that at least some of this is unnecessary, and some languages do agree here. Both [Python](https://peps.python.org/pep-0498/#specification) and [Rust](https://doc.rust-lang.org/std/macro.format.html) parse the string as a string first and thus disallow starting quote to be present in the interpolation. 

## Why I want the change

My attempt to parse this with chumsky in #1818 was along the lines of:
- parse the string char for char and switch to interpolation parsing when `{` is encountered,
- not parse char for char until `}` is encountered,
- repeat.

Result is a vector of strings each of which is either literal string or a interpolation string that still needs to be lexed and parsed separately.

The approach is not clean (we'd have to created a nested lexer and nested parser, because this produces nested streams) and also does not parse nested interpolations and double braces correctly.

## How we implement it now

This certainly is possible to parse, because we have a working impl here:

https://github.com/PRQL/prql/blob/main/prql-compiler/src/parser/prql.pest#L115-L128

I think the leading case for why this is possible is that pest parses char by char instead of lexing first. With chumsky, I opted for lexer + parser approach, see https://github.com/PRQL/prql/pull/1818#issuecomment-1427680794

## What would I change

My vote here is to simplify our grammar and drop support for having quotes within the interpolations:

```
name = 'aljaz'
f"Hello {name}"   # ok: Hello aljaz
f"Hello { 'n' }"  # ok: Hello n
f"Hello { "n" }"  # err: unexpected identifier `n` (this is parsed as string ident string)
f"Hello { '"' }"  # err: unterminated string starting at ' }"
f"Hello { '}' }"  # ok: Hello }
f"Hello {{ n }}"  # ok: Hello { n }
```

**Comment by max-sixty** [comment]
Thanks for the very clear issue! I found the examples excellent.

It's worth browsing over https://peps.python.org/pep-0701/, which proposes loosening python's rules here, closer to PRQL's current implementation. Part of their logic is that the current python parser adds _more_ complication to get their current impl, so different from us.

I don't think we'd be giving up much if we went to the stricter rules, so if it makes the compiler's implementation simpler, then I'd support. If we find that the semantics are insufficient for folks, we can always add more functionality back in the future.
